pydm/widgets/enum_button.py

```diff
@@ -127,6 +130,158 @@ def orientation(self, new_orientation):
+
+        Parameters
+        -------
+        int
+        """
+        new_margin = max(0, int(new_margin))
+        self._layout_margins.setBottom(new_margin)
+        self.layout().setContentsMargins(self._layout_margins)
+        self.rebuild_layout()
+
+    @Property(int)
+    def margin_left(self):
+        """
+        The left margin of the QGridLayout of buttons.
+
+        Returns
+        -------
+        int
+        """
+        return self._layout_margins.left()
+
+    @margin_left.setter
+    def margin_left(self, new_margin):
+        """
+        Set the left margin of the QGridLayout of buttons.
+
+        Parameters
+        -------
+        int
+        """
+        new_margin = max(0, int(new_margin))
+        self._layout_margins.setLeft(new_margin)
+        self.layout().setContentsMargins(self._layout_margins)
+        self.rebuild_layout()
+
+    @Property(int)
+    def margin_right(self):
+        """
+        The right margin of the QGridLayout of buttons.
+
+        Returns
+        -------
+        int
+        """
+        return self._layout_margins.right()
+
+    @margin_right.setter
+    def margin_right(self, new_margin):
+        """
+        Set the right margin of the QGridLayout of buttons.
+
+        Parameters
+        -------
+        int
+        """
+        new_margin = max(0, int(new_margin))
+        self._layout_margins.setRight(new_margin)
+        self.layout().setContentsMargins(self._layout_margins)
+        self.rebuild_layout()
+
+    @Property(int)
+    def horizontalSpacing(self):
+        """
+        The horizontal gap of the QGridLayout containing the QButtonGroup.
+
+        Returns
+        -------
+        int
+        """
+        return self._layout_spacing_horizontal
+
+    @horizontalSpacing.setter
+    def horizontalSpacing(self, new_spacing):
+        """
+        Set the layout horizontal gap between buttons.
+
+        Parameters
+        -------
+        new_spacing : int
+        """
+        new_spacing = max(0, int(new_spacing))
+        if new_spacing != self._layout_spacing_horizontal:
+            self._layout_spacing_horizontal = new_spacing
+            self.layout().setHorizontalSpacing(new_spacing)
+            self.rebuild_layout()
+
+    @Property(int)
+    def verticalSpacing(self):
+        """
+        The vertical gap of the QGridLayout containing the QButtonGroup.
+
+        Returns
+        -------
+        int
+        """
+        return self._layout_spacing_vertical
+
+    @verticalSpacing.setter
+    def verticalSpacing(self, new_spacing):
+        """
+        Set the layout vertical gap between buttons.
+
+        Parameters
+        -------
+        new_spacing : int
+        """
+        new_spacing = max(0, int(new_spacing))
+        if new_spacing != self._layout_spacing_vertical:
+            self._layout_spacing_vertical = new_spacing
+            self.layout().setVerticalSpacing(new_spacing)
+            self.rebuild_layout()
```

Comment:
It is not necessary to `rebuild_layout` since you are already applying the new value to the layout.
```suggestion

```